src/pkg/boltz.ts

```diff
@@ -0,0 +1,480 @@
+  blindingPublicKey: Buffer;
+  timeoutBlockHeight: number;
+}
+
+const feeAmount = 500; // fee for regular liquid tx
+const swapFeeAmount = 500; // fee for Boltz
+// lightning swap invoice amount limit (in satoshis)
+export const DEFAULT_LIGHTNING_LIMITS = { maximal: 4294967, minimal: 50000 };
+export const DEPOSIT_LIGHTNING_LIMITS = {
+  maximal: DEFAULT_LIGHTNING_LIMITS.maximal - feeAmount - swapFeeAmount,
+  minimal: DEFAULT_LIGHTNING_LIMITS.minimal - feeAmount - swapFeeAmount,
+};
+
+export const boltzUrl: Record<NetworkString, string> = {
+  regtest: 'http://localhost:9090',
+  testnet: 'https://testnet.boltz.exchange/api',
+  liquid: 'https://api.boltz.exchange',
+};
+
+export class Boltz implements BoltzInterface {
+  private asset: string;
+  private url: string;
+  private zkp: ZKP;
+
+  constructor(url: string, asset: string, zkp: ZKP) {
+    this.asset = asset;
+    this.url = url;
+    this.zkp = zkp;
+  }
+
+  async getBoltzPair(pair: string): Promise<any> {
+    const data = await this.getApi(`${this.url}/getpairs`);
+    if (!data?.pairs?.[pair]) return;
+    return data.pairs[pair];
+  }
+
+  // return invoice expire date
+  getInvoiceExpireDate(invoice: string): number {
+    const toMilliseconds = (num: number) => num * 1000;
+    const { timeExpireDate, timestamp } = bolt11.decode(invoice);
+    if (timeExpireDate) return toMilliseconds(timeExpireDate);
+    const { expire_time } = bolt11.decode(invoice).tagsObject;
+    const expireTime = toMilliseconds(expire_time ?? 3600);
+    const from = timestamp ? toMilliseconds(timestamp) : Date.now();
+    return from + expireTime;
+  }
+
+  // return value in given invoice
+  getInvoiceValue(invoice: string): number {
+    const { satoshis, millisatoshis } = bolt11.decode(invoice);
+    if (satoshis) return fromSatoshi(satoshis, 8);
+    if (millisatoshis) return fromSatoshi(Number(millisatoshis) / 1000, 8);
+    return 0;
+  }
+
+  makeClaimTransaction({
+    utxo,
+    claimKeyPair,
+    preimage,
+    redeemScript,
+    destinationScript,
+    fee,
+    blindingPublicKey,
+    timeoutBlockHeight,
+  }: MakeClaimTransactionParams): Transaction {
+    if (!utxo.blindingData) throw new Error('utxo is not blinded');
+    if (!utxo.witnessUtxo) throw new Error('utxo missing witnessUtxo');
+    const pset = Creator.newPset();
+    const updater = new Updater(pset);
+
+    pset.addInput(
+      new CreatorInput(utxo.txid, utxo.vout, 0xfffffffd, timeoutBlockHeight).toPartialInput()
+    );
+    updater.addInSighashType(0, Transaction.SIGHASH_ALL);
+    updater.addInWitnessUtxo(0, utxo.witnessUtxo);
+    updater.addInWitnessScript(0, utxo.witnessUtxo.script);
+
+    updater.addOutputs([
+      {
+        script: destinationScript,
+        blindingPublicKey,
+        asset: this.asset,
+        amount: (utxo.blindingData?.value ?? 0) - fee,
+        blinderIndex: 0,
+      },
+      {
+        amount: fee,
+        asset: this.asset,
+      },
+    ]);
+
+    console.log('pset', pset);
```

Comment:
Not a fan of the console logs in this function

Reply:
Removed